allennlp/common/testing/model_test_case.py

```diff
@@ -93,24 +94,27 @@ def ensure_model_can_train_save_and_load(
         params = Params.from_file(param_file, params_overrides=overrides)
         reader = DatasetReader.from_params(params["dataset_reader"])
 
-        # Need to duplicate params because Iterator.from_params will consume.
-        iterator_params = params["iterator"]
-        iterator_params2 = Params(copy.deepcopy(iterator_params.as_dict()))
-
-        iterator = DataIterator.from_params(iterator_params)
-        iterator2 = DataIterator.from_params(iterator_params2)
-
-        # We'll check that even if we index the dataset with each model separately, we still get
-        # the same result out.
         print("Reading with original model")
         model_dataset = reader.read(params["validation_data_path"])
-        iterator.index_with(model.vocab)
-        model_batch = next(iterator(model_dataset, shuffle=False))
+        model_dataset.index_with(model.vocab)
 
         print("Reading with loaded model")
         loaded_dataset = reader.read(params["validation_data_path"])
-        iterator2.index_with(loaded_model.vocab)
-        loaded_batch = next(iterator2(loaded_dataset, shuffle=False))
+        loaded_dataset.index_with(loaded_model.vocab)
+
+        # Need to duplicate params because DataLoader.from_params will consume.
+        data_loader_params = params["data_loader"]
+        data_loader_params["shuffle"] = False
+        data_loader_params2 = Params(copy.deepcopy(data_loader_params.as_dict()))
+
+        data_loader = DataLoader.from_params(dataset=model_dataset, params=data_loader_params)
+        data_loader2 = DataLoader.from_params(dataset=loaded_dataset, params=data_loader_params2)
+
+        # We'll check that even if we index the dataset with each model separately, we still get
+        # the same result out.
+        model_batch = next(iter(data_loader))
+
+        loaded_batch = next((iter(data_loader2)))
```

Comment:
A little thing :sweat_smile: 

```suggestion
        loaded_batch = next(iter(data_loader2))
```

Reply:
lol thanks, i've worked on this too much today